python/gigl/distributed/distributed_neighborloader.py

```
@@ -179,17 +196,28 @@ def __init__(
             pin_memory=device.type == "cuda",
         )
 
-        super().__init__(
-            data=dataset,
+        sampling_config = SamplingConfig(
+            sampling_type=SamplingType.NODE,
             num_neighbors=num_neighbors,
-            input_nodes=curr_process_nodes,
             batch_size=batch_size,
+            shuffle=False,
+            drop_last=False,
             with_edge=True,
-            edge_dir=dataset.edge_dir,
             collect_features=True,
-            to_device=device,
-            worker_options=worker_options,
+            with_neg=False,
+            with_weight=False,
+            edge_dir=dataset.edge_dir,
+            seed=None,  # it's actually optional - None means random.
         )
```

would be good to pass `**kwargs` straight to the base class for parameters that's transparent to the child class

---

I've had bad experiences with `**kwargs` in the past silently dropping stuff due to typos and I prefer to avoid them.

Also in this specific case we already pass all args to `DistLoader`. Xiaobin mentioned offline that we can pass more stuff in to `SamplingConfig` which I've added some of - but IME we should make those explicit params in the future as needed.

---

okay if we keep inheritance shallow it's good. otherwise when base class introduces a new param we need to touch every class along the inheritance chain.